jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EjbSchedulerService.java

```diff
@@ -131,15 +134,16 @@ public boolean isValid(GlobalJobHandle jobHandle) {
         return true;	    
 	}
 	
-	private String getJobName(JobContext ctx, Long id) {
+    protected String getJobName(JobContext ctx, Long id) {
 
         String jobname = null;
         
         if (ctx instanceof ProcessJobContext) {
             ProcessJobContext processCtx = (ProcessJobContext) ctx;
             jobname = processCtx.getSessionId() + "-" + processCtx.getProcessInstanceId() + "-" + processCtx.getTimer().getId();
             if (processCtx instanceof StartProcessJobContext) {
-                jobname = "StartProcess-"+((StartProcessJobContext) processCtx).getProcessId()+ "-" + processCtx.getTimer().getId();
+                String deploymentId = (String) processCtx.getKnowledgeRuntime().getEnvironment().get(EnvironmentName.DEPLOYMENT_ID);
+                jobname = deploymentId + "StartProcess-" + ((StartProcessJobContext) processCtx).getProcessId() + "-" + processCtx.getTimer().getId();
```

Comment:
Maybe for better readability when possibly debugging this, it would be good to add a hyphen, e.g. `deploymentId + "-StartProcess-"`. That's all.